refactor(linter): remove `CliRunResult` from `TsGoLintState`

Sysix opened the pull request:
Goal is there are no references for `oxlint` functions.
Now move `TsGoLintState` to `oxc_linter` so the server can use it too 🥳

camc314 commented:
### Merge activity

* **Aug 15, 12:11 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Sysix committed:
refactor(linter): remove `CliRunResult` from `TsGoLintState` (#13119)

Goal is there are no references for `oxlint` functions.
Now move `TsGoLintState` to `oxc_linter` so the server can use it too 🥳
